keylime/tpm/tpm_abstract.py

```diff
@@ -281,15 +297,34 @@ def check_pcrs(self, agent_id, tpm_policy, pcrs, data, virtual, ima_measurement_
 
                 return False
 
-            # check whether this is a MB PCR -- do *not* compare measured boot PCRs against a reference state, if one exists
-            if pcrnum in config.MEASUREDBOOT_PCRS and mb_intended_state :
+            # PCRs set by measured boot get their own special handling
+
+            if pcrnum in config.MEASUREDBOOT_PCRS :
+
+                if mb_refstate not in (False, None):
+
+                    if mb_measurement_list in (False, None) :
```

Comment:
Same as above

Reply:
there was reason for this (we discussed the possibility of admitting an empty dictionary as a reference state), but decided against if for this PR